python/lsst/dax/obscore/plugins.py

```diff
@@ -0,0 +1,59 @@
+# This file is part of dax_obscore.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ["get_siav2_handler"]
+
+from importlib.metadata import EntryPoint, entry_points
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from .siav2 import SIAv2Handler
+
+
+def _get_siav2_entry_points() -> dict[str, EntryPoint]:
+    plugins = entry_points(group="dax_obscore.siav2")
+    return {p.name: p for p in plugins}
+
+
+def get_siav2_handler(namespace: str) -> type[SIAv2Handler]:
+    """Select the correct handler for this universe namespace.
+
+    Parameters
+    ----------
+    namespace : `str`
+        The butler dimension universe namespace.
+
+    Returns
+    -------
+    handler : `type` [ `lsst.dax.obscore.siav2.SIAv2Handler` ]
+        The class of handler suitable for this namespace.
+    """
+    plugins = _get_siav2_entry_points()
+    entry_point = plugins.get(namespace)
+    if entry_point is None:
+        known = ", ".join(plugins.keys())
+        raise RuntimeError(
+            f"Unable to find suitable SIAv2 Handler for namespace {namespace} [do understand: {known}]"
+        )
+    func = entry_point.load()
+    return func()
```

Comment:
Maybe worth checking the type of the returned class?

Reply:
Now I remember why I didn't do it. You get a circular import error because siav2.py uses the plugins api and so the plugins api can't also load siav2. I'll do a delayed import (which should be a no-op anyhow after the `load` has been called.